### PR TITLE
[rmkit] Add RMKIT_DEFAULT_FONT env var

### PR DESCRIPTION
**Pull request by mrichards42**
Mostly adding this for resim builds, but I'm not attached to the idea or the name.

### CODE REVIEW

**Comment by raisjn** [comment]
i guess people are getting burned right and left by this, should also print error when file isn't there.

**Review by raisjn — APPROVED** [review]

**Comment by raisjn** [comment]
btw, i'm using fork of resim: https://pypi.org/project/rmkit-sim/ (`pip install rmkit-sim`) which removes most of the buttons and has slightly better scaling algorithm (using PIL) and supports scaling with floats

**Comment by mrichards42** [comment]
> should also print error when file isn't there.

good call -- added.

Thanks for the updated resim pointer too, it's definitely better!

File: src/rmkit/fb/stb_text.cpy [hunk 1]
```diff
@@ -17,13 +17,20 @@ namespace stbtext:
...
+      if filename == NULL:
+        #ifdef REMARKABLE
+        filename = "/usr/share/fonts/ttf/noto/NotoMono-Regular.ttf";
+       // TODO: fix the max size read to prevent overflows (or just abort on really large files)
+        #else
+        filename = "src/vendor/NotoSansMono-Regular.ttf";
+        #endif
+      FILE * file = fopen(filename, "rb");
+      if file == NULL:
+        cerr << "Unable to read font file: " << filename << endl;
```
Comment (raisjn) [inline src/rmkit/fb/stb_text.cpy:30]: debug

**Review by raisjn — APPROVED** [review]
use debug instead of cerr

**Comment by raisjn** [comment]
thank you, if i approve changes i don't mind you updating and merging, pls rerequest only if you think i should look closely, otherwise i trust you

**Comment by okayzed** [comment]
interestingly, this does not compile on CI or on my machine - need to figure out what happened

**Comment by raisjn** [comment]
broken build should be fixed in https://github.com/rmkit-dev/rmkit/commit/6c26f3e76d8130351b60c7e4e7c557d1de974e7e. okp has problem with dangling `#` statements

**Comment by mrichards42** [comment]
shoot, sorry I didn't notice that broke the build! the dangling `#` would explain something I was having trouble with earlier though, so thanks for explaining that.

**Comment by raisjn** [comment]
https://github.com/raisjn/okp/commit/2af25e08786378cedad7d0ddb77e40be066386bd should fix the dangling hash lines and is released as 0.0.50

**Pull request merged**